repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapper.scala

```diff
@@ -145,4 +151,71 @@ class HttpServletRequestWrapper(originalRequest: HttpServletRequest, inputStream
   def getSplittableHeaderScala(headerName: String): List[String] = getHeadersScala(headerName).foldLeft(List.empty[String])((list, s) => list ++ s.split(","))
 
   override def getSplittableHeaders(headerName: String): util.List[String] = getSplittableHeaderScala(headerName).asJava
+
+  /**
+    * @return a [[StringBuffer]] containing the reconstructed URL for this request (note that
+    *         mutation of this [[StringBuffer]] will have no effect on the request URL)
```

Comment:
Thought I'd point out this comment. The JavaDoc states:
"Because this method returns a StringBuffer, not a string, you can modify the URL easily, for example, to append query parameters."

However, that seems altogether like a bad idea. I'm not sure why the contract specifies a StringBuffer, to be honest.